src/overmind/creditsHistory/effects.ts

```diff
@@ -0,0 +1,88 @@
+
+export const getCreditsHistory = async (): Promise<EffectResponse<IssuedRetiredGraphData>> => {
+  return new Promise((resolve) => {
+    let result: EffectResponse<IssuedRetiredGraphData>
+    axios
+      .get(`${defaultDomain}/widgets/issued-retired-graph`, defaultHeaders)
+      .then((body) => {
+        if (body.data && body.data.lastUpdated && body.data.countriesData) {
+          const mapData = body.data as IssuedRetiredGraphData
+
+          result = { data: mapData }
+        } else {
+          result = { error: { code: body.status.toString(), message: body.statusText } }
+        }
+        resolve(result)
+      })
+      .catch((_) => {
+        result = { error: { code: `400`, message: `could not fetch data` } }
+        resolve(result)
+      })
+  })
+}
+
+export const appendChartDataAndStat = (result: FilteredCreditsHistoryData, formattedDateTime: Date, issued: number, retired: number) => {
+  const index = result.chartData[0].data.findIndex((e) => e.x === formattedDateTime)
+  if (index < 0) {
+    result.chartData[0].data.push({ x: formattedDateTime, y: issued })
+    result.chartData[1].data.push({ x: formattedDateTime, y: retired })
+  } else {
+    result.chartData[0].data[index] = { x: formattedDateTime, y: result.chartData[0].data[index].y + issued }
+    result.chartData[1].data[index] = { x: formattedDateTime, y: result.chartData[1].data[index].y + retired }
+  }
+  result.issued += issued
+  result.retired += retired
+}
+
+export const generateFilteredCreditsHistory = (rawData: IssuedRetiredDataCountry[], dataFilters: DataFilters): FilteredCreditsHistoryData => {
+  const today = startOfMonth(new Date())
+  const { country, region, timeframe } = dataFilters
+
+  const result: FilteredCreditsHistoryData = {
+    chartData: [
+      {
+        id: `issued`,
+        data: [],
+      },
+      {
+        id: `retired`,
+        data: [],
+      },
+    ],
+    issued: 0,
+    retired: 0,
+  }
+
+  rawData.forEach((countryEntry) => {
+    if (region === SubRegion.WORLD || (country == undefined && generateCountryByRegion(region).includes(countryEntry.countryCode)) || countryEntry.countryCode == country) {
+      countryEntry.timeRanges.forEach((timeRangeEntry) => {
+        const { year, retired, issued, month } = timeRangeEntry
+        const formattedDateTime = new Date(year, month - 1)
+        if (
+          timeframe === TimeframesData.MAX ||
+          // eslint-disable-next-line no-magic-numbers
+          (timeframe === TimeframesData.ONE_YEAR && differenceInMonths(today, formattedDateTime) <= 12) ||
+          // eslint-disable-next-line no-magic-numbers
+          (timeframe === TimeframesData.SIX_MONTHS && differenceInMonths(today, formattedDateTime) <= 6) ||
+          (timeframe === TimeframesData.ONE_MONTH && differenceInMonths(today, formattedDateTime) <= 1)
+        ) {
+          appendChartDataAndStat(result, formattedDateTime, issued, retired)
```

Comment:
These numbers could be in const or enum this would stop the use of `// eslint-disable-next-line no-magic-numbers`